node/src/blockchain/blockchain_bridge.rs

```diff
@@ -400,33 +401,32 @@ impl BlockchainBridge {
                 .process_transaction_receipts(transaction_hashes)
                 .map_err(move |e| e.to_string())
                 .and_then(move |transaction_receipts_results| {
-                    let length = transaction_receipts_results.len();
-                    let mut transactions_found = 0;
-                    for transaction_receipt in &transaction_receipts_results {
-                        if let TransactionReceiptResult::RpcResponse(tx_receipt) = transaction_receipt {
-                            if let TxStatus::Succeeded(_) = tx_receipt.status {
-                                transactions_found += 1;
+                    let (successful_count, failed_count, pending_count) = transaction_receipts_results
+                        .iter()
+                        .fold((0, 0, 0), |(success, fail, pending), transaction_receipt| {
+                            match transaction_receipt {
+                                TransactionReceiptResult::RpcResponse(tx_receipt) => match tx_receipt.status {
+                                    TxStatus::Failed => (success, fail + 1, pending),
+                                    TxStatus::Pending => (success, fail, pending + 1),
+                                    TxStatus::Succeeded(_) => (success + 1, fail, pending),
+                                },
+                                TransactionReceiptResult::LocalError(_)=> (success, fail, pending + 1),
                             }
-                        }
-                    }
+                        });
+                    debug!(logger, "Scan results: Successful: {successful_count}, Pending: {pending_count}, Failed: {failed_count}");
```

Comment:
Do you think you could pack this whole fold for the debug into a separate function. 

I've got this in my mind: 

You can use the trick with 
```
logger.debug(||{
 <your function returns a String (the debug message)>
})
```

Therefore: 

```
fn transaction_receipts_results_debug(transaction_receipts_results: <The proper type>){
       logger.debug({
              let (successful_count, failed_count, pending_count) = transaction_receipts_results
                       .iter()
                       .fold((0, 0, 0), |(success, fail, pending), transaction_receipt| {
                               match transaction_receipt {
                                    TransactionReceiptResult::RpcResponse(tx_receipt) => match tx_receipt.status {
                                        TxStatus::Failed => (success, fail + 1, pending),
                                        TxStatus::Pending => (success, fail, pending + 1),
                                        TxStatus::Succeeded(_) => (success + 1, fail, pending),
                                    },
                                   TransactionReceiptResult::LocalError(_)=> (success, fail, pending + 1),
                              }
              format!("Scan results: Successful: {successful_count}, Pending: {pending_count}, Failed: {failed_count}");
      })
}
```

A praise for that nice use of `.fold()`. :wink: